### DISCUSSION TITLE
Is there a limit on how many nodes can be subscribed to?

### DISCUSSION THREAD

**Discussion by ctsaitsao** [FreeOpcUa/python-opcua#1381] [Q&A] [answered]
Hi, I am encountering issues receiving data from a few PLC nodes that are added later on during my program and would like to know if there's a limit to how many nodes can be fed to the `subscribe_data_change()` method or if there are ways to avoid possible race conditions. I am not sure how `datachange_notification()` works under the hood so an explanation of that would help as well.

**Comment by AndreasHeine** [comment, marked as answer]
yes a serverside limit! s7 for example have 10k to 20k max monitored items per subscription depending on the model could also be lower.

**Reply by ctsaitsao** [reply]
Hm, I definitely wouldn't be exceeding such a limit. 

However, I am calling `subscribe_data_change()` multiple times and was wondering if this can lead to duplicate subscriptions being created (basically, I call it every time I want to get data from a new node).

Thanks for answering!

**Reply by AndreasHeine** [reply]
no it will just add a monitoreditem to the subscription!
